### DISCUSSION TITLE
Connect Locally installed VS code to remotely hosted code server

### DISCUSSION THREAD

**Discussion by ksbalajee** [coder/code-server#5211] [General]
Hi,

Is there anyway that I can connect my local VS code to point to the remotely hosted code server and use it like that instead of using the VS code in browser?

**Comment by jsjoeio** [comment]
I think you should be able to use SSH and [this extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh)

**Reply by craftxbox** [reply]
I tried this in reverse (ie, code-server to remote machine rather than the other way around), Remote-SSH doesn't wish to enable because of its ui-only extensionKind, however, even when manually removing that, It will only end up reloading the current window (or, creating a new window) and nothing else of substance. Otherwise, other parts of the extension appear to function as intended?

**Reply by ksbalajee** [reply]
@jsjoeio Remote ssh extension, sets us the Microsoft 's vscode server on the host that I'm getting connected to and connects the locally installed vscode to that server. I think I'll explore the Remote ssh extension and make some changes to it such that it connects to the code server that I hosted instead of setting up and connecting to its own server.

**Reply by ksbalajee** [reply]
Oops, Remote SSH and github codespace extensions aren't available as open source. Looks like we need to build an extension similar to Github codespaces to be able to connect with our instance of code server to connect from a local VS  code.

**Reply by jsjoeio** [reply]
Ah. I forgot they weren't open source. Yeah, I think this has been discussed in the past but no one has taken on the challenge. 

Some previous discussion: https://github.com/coder/code-server/issues/1315

**Reply by ksbalajee** [reply]
Thanks @jsjoeio,
I've just walked through the entire thread. I'll also try to RE and post my findings if there's a breakthrough

**Reply by HewwoBerry** [reply]
Did you end up finding anything that is usable?
